cdap-security/src/test/java/co/cask/cdap/security/authorization/AuthorizationUtilTest.java

```diff
@@ -0,0 +1,92 @@
+import org.junit.BeforeClass;
+
+import java.net.InetAddress;
+
+/**
+ * Test for {@link AuthorizationUtil}
+ */
+public class AuthorizationUtilTest {
+  private static CConfiguration cConf;
+  private static InMemoryNamespaceClient namespaceClient;
+  private static AuthenticationContext authenticationContext;
+  private static final NamespaceId namespaceId = new NamespaceId("AuthorizationUtilTest");
+  private static final ApplicationId applicationId = namespaceId.app("someapp");
+  private static String username;
+
+  @BeforeClass
+  public static void init() throws Exception {
+    cConf = CConfiguration.create();
+    // Note: it is important to initialize the UGI before we call AuthorizationUtil.getAppAuthorizingUser(..)
+    // which uses KerberosName since KerberosName expect the rules for matching to be set. See
+    // http://lucene.472066.n3.nabble.com/KerberosName-rules-are-null-during-KerberosName-getShortName-
+    // in-KerberosAuthenticationHandler-td4074309.html for more context
+    username = UserGroupInformation.getCurrentUser().getShortUserName();
+    namespaceClient = new InMemoryNamespaceClient();
+    authenticationContext = new AuthenticationTestContext();
+  }
+
+  @org.junit.Test
+  public void testGetAppAuthorizingUse() throws Exception {
+    OwnerAdmin ownerAdmin = getOwnerAdmin();
+    // test with complete principal (alice/somehost.net@somerealm.net)
+    String principal = username + "/" + InetAddress.getLocalHost().getHostName() + "@REALM.net";
+    NamespaceMeta nsMeta = new NamespaceMeta.Builder().setName(namespaceId).setPrincipal(principal)
+      .setKeytabURI("doesnotmatter").build();
+    namespaceClient.create(nsMeta);
+    Assert.assertEquals(username, AuthorizationUtil.getAppAuthorizingUser(ownerAdmin, authenticationContext,
+                                                                          applicationId, null));
+
+    // test with principal which is just username (alice)
+    namespaceClient.delete(namespaceId);
+    principal = username;
+    nsMeta = new NamespaceMeta.Builder().setName(namespaceId).setPrincipal(principal)
+      .setKeytabURI("doesnotmatter").build();
+    namespaceClient.create(nsMeta);
+    Assert.assertEquals(username, AuthorizationUtil.getAppAuthorizingUser(ownerAdmin, authenticationContext,
+                                                                          applicationId, null));
+
+    // test with principal and realm (alice@somerealm.net)
+    namespaceClient.delete(namespaceId);
+    principal = username + "@REALM.net";
+    nsMeta = new NamespaceMeta.Builder().setName(namespaceId).setPrincipal(principal)
+      .setKeytabURI("doesnotmatter").build();
+    namespaceClient.create(nsMeta);
+    Assert.assertEquals(username, AuthorizationUtil.getAppAuthorizingUser(ownerAdmin, authenticationContext,
+                                                                          applicationId, null));
+    // clean up
+    namespaceClient.delete(namespaceId);
+  }
+
```

Comment:
you should also test the case where the authentication context has a principal of each of the three forms (and the namespace has no principal). AuthenticationTestContext only provides short username as principal.

Reply:
I don't think AuthenticationContext can be set by kerberos principal. Its set with the logged in username from our authentication module. We don't support logging in with kerberos principal as of now.